stns.c

```diff
@@ -289,13 +288,13 @@ static CURLcode inner_http_request(stns_conf_t *c, char *path, stns_response_t *
 
   result = curl_easy_perform(curl);
 
-  if (result != CURLE_OK) {
+  long code;
+  curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
+  if (code >= 400) {
     syslog(LOG_ERR, "%s(stns)[L%d] http request failed: %s", __func__, __LINE__, curl_easy_strerror(result));
-    if (result == CURLE_HTTP_RETURNED_ERROR) {
-      long code;
-      curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
-      res->status_code = code;
```

Comment:
why don't you set status code when return code over 4xx?

Reply:
sorry, simply i forgot to set :bow: